src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -341,11 +341,10 @@ public override uint EnableSsl(uint options)
         /// </summary>
         public override void DisableSsl()
         {
-#if MONO
-            // Temp workaround - SSLStream.Dispose causes an expected behavior in mono for windows
-            if (Environment.OSVersion.Platform != PlatformID.Win32NT)
-#endif
+#if !MONO || MONO_FEATURE_BTLS || MONO_FEATURE_APPLETLS
+            // Temp workaround - SSLStream.Dispose causes an expected behavior with legacy ssl implementation
```

Comment:
I guess the comment should be `SSLStream.Dispose causes an unexpected behaviour...`